### ISSUE TITLE
Optimize for local RPC

### ISSUE THREAD

**Issue by TennyZhuang** [singularity-data/risingwave#3443]
We always issue a gRPC request even if they are in the same process. Is it possible and worth introducing a trait and optimizing for the local request?

The task is marked as `difficulty/hard`, before implementation, **an RFC is required.**

**Comment by TennyZhuang** [comment]
One use case is that we may integrate compactor/frontend/compute node in one process for PoC.

**Comment by xiangjinwu** [comment]
FYI
https://github.com/singularity-data/risingwave/blob/68be2299764ae71c39a20935c9591b9c9cf1f515/src/batch/src/executor/generic_exchange.rs#L70-L88

**Comment by TennyZhuang** [comment]
> FYI
> 
> https://github.com/singularity-data/risingwave/blob/68be2299764ae71c39a20935c9591b9c9cf1f515/src/batch/src/executor/generic_exchange.rs#L70-L88

Yes, I know there is special path for exchange, just thinking about can we unify it.
